python/ray/workers/setup_runtime_env.py

```diff
@@ -20,11 +21,15 @@ def setup(input_args):
     commands = []
     runtime_env: RuntimeEnvDict = json.loads(args.serialized_runtime_env
                                              or "{}")
+
+    py_executable: str = sys.executable
+
     if runtime_env.get("conda"):
         if isinstance(runtime_env["conda"], str):
             commands += get_conda_activate_commands(runtime_env["conda"])
+            py_executable = "python"
```

Comment:
When a conda environment is activated, `python`will point to the correct executable for that conda environment.

Reply:
Conda activate will make the bash to find the correct Python. But, It's dangerous to run bash with user inputs. For example, user may write a special conda name to remove all data from all nodes.

Reply:
Ah thanks, that's a really good point.  I wonder what would be a good way to validate the conda name input. One thought I had was to simply run "conda activate <name>" in a subprocess and check if it fails, but that suffers from the same problem you mentioned.  I made an issue for it here https://github.com/ray-project/ray/issues/15837

Reply:
> Ah thanks, that's a really good point. I wonder what would be a good way to validate the conda name input. One thought I had was to simply run "conda activate " in a subprocess and check if it fails, but that suffers from the same problem you mentioned. I made an issue for it here #15837

This PR use the `pydantic` to validate the RuntimeEnv. https://github.com/ray-project/ray/pull/15804